homeassistant/components/goodwe/sensor.py

```diff
@@ -0,0 +1,262 @@
+        {vol.Required(ATTR_ONGRID_BATTERY_DOD): vol.Coerce(int)},
+        "async_set_ongrid_battery_dod",
+    )
+    platform.async_register_entity_service(
+        SERVICE_SET_GRID_EXPORT_LIMIT,
+        {vol.Required(ATTR_GRID_EXPORT_LIMIT): vol.Coerce(int)},
+        "async_set_grid_export_limit",
+    )
+
+    return True
+
+
+class InverterEntity(CoordinatorEntity, SensorEntity):
+    """Entity representing the inverter instance itself."""
+
+    _MAIN_ENTITY_SENSOR = "ppv"
+
+    def __init__(
+        self,
+        coordinator: DataUpdateCoordinator,
+        device_info: DeviceInfo,
+        inverter: Inverter,
+    ) -> None:
+        """Initialize the main inverter entity."""
+        super().__init__(coordinator)
+        self.entity_id = f".{DOMAIN}_inverter"
+        self._attr_unique_id = f"{DOMAIN}-{inverter.serial_number}"
+        self._attr_device_info = device_info
+        self._attr_icon = "mdi:solar-power"
+        self._attr_name = "PV Inverter"
+        self._attr_native_unit_of_measurement = POWER_WATT
+        self._attr_extra_state_attributes = {"serial_number": inverter.serial_number}
+        self._inverter: Inverter = inverter
+
+    async def async_set_work_mode(self, work_mode: int) -> None:
+        """Set the inverter work mode."""
+        await self._inverter.set_work_mode(work_mode)
+
+    async def async_set_ongrid_battery_dod(self, ongrid_battery_dod: int) -> None:
+        """Set the on-grid battery dod."""
+        await self._inverter.set_ongrid_battery_dod(ongrid_battery_dod)
+
+    async def async_set_grid_export_limit(self, grid_export_limit: int) -> None:
+        """Set the grid export limit."""
+        await self._inverter.set_grid_export_limit(grid_export_limit)
+
+    @property
+    def available(self) -> bool:
+        """Return True if entity is available."""
+        return self.coordinator.data is not None
+
+    @property
+    def native_value(self):
+        """Return the value reported by the sensor."""
+        if self.coordinator.data is not None:
+            new_value = self.coordinator.data.get(self._MAIN_ENTITY_SENSOR)
+            # If no new value was provided, keep the previous
+            if new_value is not None:
+                self._attr_native_value = new_value
+
+        return self._attr_native_value
+
+    @property
+    def state_attributes(self):
+        """Return the inverter state attributes."""
+        return self.coordinator.data
+
+
+class InverterSensor(CoordinatorEntity, SensorEntity):
+    """Entity representing individual inverter sensor."""
+
+    def __init__(
+        self,
+        coordinator: DataUpdateCoordinator,
+        device_info: DeviceInfo,
+        inverter: Inverter,
+        sensor: Sensor,
+    ) -> None:
+        """Initialize an inverter sensor."""
+        super().__init__(coordinator)
+        self.entity_id = f".{DOMAIN}_{sensor.id_}"
+        self._attr_unique_id = f"{DOMAIN}-{sensor.id_}-{inverter.serial_number}"
```

Comment:
I'm not sure, there are users out there with more than 1 year of data collected, this change would break their existing stuff, so I prefer to keep it this way.

Reply:
I think that `self.entity_id` is used for data collecting.

Reply:
Yes, but changing unique_id would cause new entity_id (e.g. with suffix 2) unless I'm mistaken.

Reply:
I mean this one will be probably called https://github.com/home-assistant/core/blob/c0a8cea6fbcb533079ed429647a650ea30a7a769/homeassistant/util/__init__.py#L112

Reply:
It may be possible to modify existing entities: https://github.com/home-assistant/core/blob/c0a8cea6fbcb533079ed429647a650ea30a7a769/homeassistant/helpers/entity_registry.py#L419

Reply:
A migration would be great, but maybe keeping the existing unique_id is easier. I suggest you wait for a "master" reviewer before changing this.